### PR TITLE
Core/hubot adapers

### PR DESCRIPTION
**Pull request by mrkno**

### CODE REVIEW

File: core/modules/hubotModule.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            }
+            h.push(l);
+        }
+        return h;
+    };
+};
+
+exports.loadModule = function (module) {
+    if (!global.coffeescriptLoaded && module.startup.endsWith('.coffee')) {
+        require("coffee-script").register();
```
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l117)

File: core/modules/hubotModule.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    modulesDir = 'modules',
+    descriptor = 'hubot.json',
+    pkg = 'package.json',
+    Robot = require.once('./hubot/robot.js');
+
+var verifyModuleDescriptior = function (hj, disabled) {
+    if (!hj.name || !hj.startup || !hj.version) {
+        return false;
+    }
+
```
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l25)

File: core/modules/hubotModule.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    pkg = 'package.json',
+    Robot = require.once('./hubot/robot.js');
+
+var verifyModuleDescriptior = function (hj, disabled) {
+    if (!hj.name || !hj.startup || !hj.version) {
+        return false;
+    }
+    
+    if (disabled === true && exports.disabledConfig 
+        && exports.disabledConfig[hj.name] && exports.disabledConfig[hj.name] === true) {
```
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: ['&&' should be placed at the end of the line.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l27)

File: core/modules/hubotModule.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        desc = path.join(folderPath, './' + descriptor),
+        pack = path.join(folderPath, './' + pkg),
+        hj;
+
+    try {
+        fs.statSync(desc);
+        hj = require.once(desc);
+    }
+    catch (e) {
+        var l;
```
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: ["l" is defined but never used](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l49)

File: core/modules/hubotModule.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        } catch (e) {
+            console.debug('A failure occured while listing "' + data[i] + '". It doesn\'t appear to be a module.');
+            console.critical(e);
+            continue;
+        }
+    }
+    return modules;
+};
+
+var createHelp = function (module) {
```
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: ["createHelp" is defined but never used](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l101)

File: core/modules/hubotModule.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
+﻿/**
```
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/modules/hubotModule.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    descriptor = 'hubot.json',
+    pkg = 'package.json',
+    Robot = require.once('./hubot/robot.js');
+
+var verifyModuleDescriptior = function (hj, disabled) {
+    if (!hj.name || !hj.startup || !hj.version) {
+        return false;
+    }
+    
+    if (disabled === true && exports.disabledConfig
```
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l26)

File: core/modules/hubotModule.js (hunk 4 above)
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: ['l' is defined but never used.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l49)

File: core/modules/hubotModule.js (hunk 5 above)
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: ['createHelp' is defined but never used.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l101)

File: core/modules/hubotModule.js (hunk 1 above)
Comment (mrkno) [inline core/modules/hubotModule.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530854170/issues/source?bid=3282987&fileBranchId=3283570#l117)

File: core/integrations/integrations.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+    if (started) {
+        throw 'Cannot start when already started.';
+    }
+
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        try {
+            console.write("Loading integration '" + selectedIntegrations[i].name + "'...\t");
+            selectedIntegrations[i].instance.start(callback);
+            console.info("[DONE]");
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l111)

File: core/integrations/integrations.js [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    if (!started) {
+        throw 'Cannot stop integrations if they haven\'t been started.';
+    }
+
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        try {
+            selectedIntegrations[i].instance.stop();
+        }
+        catch (e) {
+            console.debug("Failed to correctly stop output integration '" + selectedIntegrations[i] + "'.");
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l133)

File: core/integrations/integrations.js [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        try {
+            console.write("Loading integration '" + selectedIntegrations[i].name + "'...\t");
+            selectedIntegrations[i].instance.start(callback);
+            console.info("[DONE]");
+        }
+        catch (e) {
+            console.error("[FAIL]");
+            console.debug("Failed to start output integration '" + selectedIntegrations[i].name + "'.");
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l115)

File: core/integrations/integrations.js [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+        throw 'Integrations must be set before starting';
+    }
+
+    if (started) {
+        throw 'Cannot start when already started.';
+    }
+
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        try {
+            console.write("Loading integration '" + selectedIntegrations[i].name + "'...\t");
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l109)

File: core/integrations/integrations.js [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+
+exports.setIntegrationConfigs = function(platform) {
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        selectedIntegrations[i].instance.platform = platform;
+        selectedIntegrations[i].instance.config = platform.config.loadOutputConfig(selectedIntegrations[i].name);
+        if (!selectedIntegrations[i].instance.config.commandPrefix) {
+            selectedIntegrations[i].instance.config.commandPrefix = platform.defaultPrefix;
+        }
+    }
+}
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l67)

File: core/integrations/integrations.js [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+exports.setIntegrations = function (integrations) {
+    if (selectedIntegrations) {
+        throw 'Cannot change integrations when they are already set.';
+    }
+
+    var i = 0;
+    try {
+        for (i = 0; i < integrations.length; i++) {
+            if (!global.coffeescriptLoaded && integrations[i].start.endsWith('.coffee')) {
+                require("coffee-script").register();
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l78)

File: core/integrations/integrations.js [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        try {
+            console.write("Loading integration '" + selectedIntegrations[i].name + "'...\t");
+            selectedIntegrations[i].instance.start(callback);
+            console.info("[DONE]");
+        }
+        catch (e) {
+            console.error("[FAIL]");
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l114)

File: core/integrations/integrations.js (hunk 12 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l67)

File: core/integrations/integrations.js (hunk 8 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l111)

File: core/integrations/integrations.js (hunk 9 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l133)

File: core/shim.js [hunk 15]
```diff
@@ -58,6 +58,12 @@ exports.createPlatformModule = function(platform) {
 			platform.sendMessage('If I could set the title of this chat I would set it to "' + title + '"', thread);
 		}
 	}
+
```
Comment (Awarua-) [inline core/shim.js]: Copy of function

File: core/platform.js [hunk 16]
```diff
@@ -12,13 +12,13 @@
 
 var figlet = require('figlet');
 
-Platform = function(modes) {
+Platform = function () {
```
Comment (Awarua-) [inline core/platform.js]: white space

File: main.js [hunk 17]
```diff
@@ -27,33 +27,40 @@ require('./core/prototypes.js');
...
     argp = require('./core/arguments.js'),
-    modes = modesf.listModes(),
     args = process.argv;
 
 args.splice(0, 2);
 
 // Parse optional arguments
 argp.runArguments(args);
 
-// Check startup modes
```
Comment (Awarua-) [inline main.js]: Put comment back

File: core/integrations/integrations.js [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+                    start: path.join(p, desc.main)
+                });
+            } catch (e) {
+                console.error('Invalid integration installed (syntax error?).');
+                console.critical(e);
+            }
+        }
+    }
+    cachedIntegrations = list;
+	return list;
```
Comment (Awarua-) [inline core/integrations/integrations.js]: indentation

File: core/config.js [hunk 19]
```diff
@@ -123,9 +123,9 @@ exports.loadOutputConfig = function (outputName) {
     if (!config[outputName]) {
         config[outputName] = {};
     }
-	
+
     for (var obj in config) {
-        if (modes.includes(obj)) continue;
+        if (integs.find(int => int.name === obj)) continue;
```
Comment (mrkno) [inline core/config.js]: Codacy Issue found: [Expected { after 'if' condition.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928279/issues/source?bid=3282987&fileBranchId=3283570#l128)

Comment (mrkno) [inline core/config.js]: Codacy Issue found: [Expected '{' and instead saw 'continue'.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928279/issues/source?bid=3282987&fileBranchId=3283570#l128)

File: core/integrations/integrations.js (hunk 12 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l67)

File: core/integrations/integrations.js (hunk 8 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l111)

File: core/integrations/integrations.js (hunk 9 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l133)

File: core/integrations/integrations.js [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+    started                 = false;
+
+exports.listIntegrations = function () {
+    if (cachedIntegrations) {
+        return cachedIntegrations;
+    }
+    var integ = files.filesInDirectory('./' + integrationsLocation),
+        list = [];
+    for (var i = 0; i < integ.length; i++) {
+        var p = path.join(__dirname, '../../'+ integrationsLocation, integ[i]);
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l27)

File: core/integrations/integrations.js (hunk 13 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l78)

File: core/integrations/integrations.js (hunk 11 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l109)

File: core/integrations/integrations.js [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+            selectedIntegrations[i].instance.stop();
+        }
+        catch (e) {
+            console.debug("Failed to correctly stop output integration '" + selectedIntegrations[i] + "'.");
+            console.critical(e);
+        }
+    }
+
+    started = false;
+};
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l139)

File: core/integrations/integrations.js (hunk 10 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l115)

File: core/startup.js [hunk 22]
```diff
@@ -31,11 +30,6 @@ var selectedModes    = null,
     },
     checkShutdownCode = function(code) {
         if (code === StatusFlag.ShutdownShouldRestart) {
-            startNewPlatform();
+            exports.run();
         }
     };
```
Comment (mrkno) [inline core/startup.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929747/issues/source?bid=3282987&fileBranchId=3283570#l35)

File: core/config.js (hunk 19 above)
Comment (mrkno) [inline core/config.js]: Codacy Issue found: [Expected { after 'if' condition.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929716/issues/source?bid=3282987&fileBranchId=3283570#l128)

Comment (mrkno) [inline core/config.js]: Codacy Issue found: [Expected '{' and instead saw 'continue'.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929716/issues/source?bid=3282987&fileBranchId=3283570#l128)

File: core/integrations/hubot/robot.js [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+    data: exports.config,
+    userForId: function() {
+        return {
+            id: 'foo',
+            name: 'bar'
+        }
+    }
+};
+
+exports.name = "Kassy";
```
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929734/issues/source?bid=3282987&fileBranchId=3283570#l52)

Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929734/issues/source?bid=3282987&fileBranchId=3283570#l52)

File: core/integrations/hubot/robot.js [hunk 24]
```diff
@@ -0,0 +1,52 @@
+﻿var instance,
```
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929734/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/integrations/hubot/robot.js [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+    return exports;
+};
+
+exports.brain = {
+    data: exports.config,
+    userForId: function() {
+        return {
+            id: 'foo',
+            name: 'bar'
+        }
```
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929734/issues/source?bid=3282987&fileBranchId=3283570#l48)

File: core/integrations/hubot/robot.js (hunk 23 above)
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929734/issues/source?bid=3282987&fileBranchId=3283570#l52)

File: core/integrations/hubot/robot.js (hunk 25 above)
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929734/issues/source?bid=3282987&fileBranchId=3283570#l48)

File: core/integrations/integrations.js (hunk 20 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530929735/issues/source?bid=3282987&fileBranchId=3283570#l27)

File: core/integrations/hubot/robot.js (hunk 24 above)
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2530937698/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/integrations/integrations.js (hunk 20 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530937699/issues/source?bid=3282987&fileBranchId=3283570#l27)

File: core/integrations/integrations.js [hunk 26]
```diff
@@ -0,0 +1,139 @@
+﻿/**
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2530937699/issues/source?bid=3282987&fileBranchId=3283570#l1)

**Comment by Awarua-** [comment]
Approved

File: core/integrations/integrations.js (hunk 26 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2530938252/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/integrations/hubot/robot.js (hunk 24 above)
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2530938251/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/integrations/integrations.js [hunk 27]
```diff
@@ -0,0 +1,146 @@
...
+    return list;
+};
+
+exports.setIntegrationConfigs = function(platform) {
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        selectedIntegrations[i].instance.platform = platform;
+        selectedIntegrations[i].instance.config = platform.config.loadOutputConfig(selectedIntegrations[i].name);
+
+        for (var name in selectedIntegrations[i].instance.config) {
+            if (name.startsWith("ENV_") && name === name.toUpperCase()) {
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531155566/issues/source?bid=3282987&fileBranchId=3283570#l65)

File: core/integrations/integrations.js [hunk 28]
```diff
@@ -0,0 +1,146 @@
+﻿/**
```
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2531155566/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/integrations/integrations.js (hunk 27 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531155566/issues/source?bid=3282987&fileBranchId=3283570#l65)

File: core/integrations/hubot/robot.js (hunk 24 above)
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2531155565/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/core_integrations/test.js [hunk 29]
```diff
@@ -0,0 +1,78 @@
...
+        exports.config.senderName = 'TESTING';
+    }
+    if (!exports.config.commandHistory) {
+        exports.config.commandHistory = [];
+    }
+
+    api = api = shim.createPlatformModule({
+        commandPrefix: exports.config.commandPrefix,
+        sendMessage: function (text) {
+            console.log(chalk.bold("" + text));
```
Comment (mrkno) [inline core/core_integrations/test.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315428/issues/source?bid=3282987&fileBranchId=3283570#l29)

File: core/core_integrations/test.js [hunk 30]
```diff
@@ -0,0 +1,78 @@
...
+    }, 100);
+};
+
+exports.stop = function () {
+    hasStartedShutdown = true;
+    cli.close();
+    if (exports.config.commandHistory.length > historySize) {
+        exports.config.commandHistory.splice(0, exports.config.commandHistory.length - historySize);
+    }
+};
```
Comment (mrkno) [inline core/core_integrations/test.js]: Codacy Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315428/issues/source?bid=3282987&fileBranchId=3283570#l78)

File: core/core_integrations/test.js [hunk 31]
```diff
@@ -0,0 +1,78 @@
...
+
+    cli.command('history', function () {
+        var hist = exports.config.commandHistory,
+            results = '';
+        for (var i = 0; i < hist.length; i++) {
+            results += hist[i] + (i + 1 === hist.length ? '' : '\n');
+        }
+        return api.sendMessage(results, exports.config.thread);
+    });
+
```
Comment (mrkno) [inline core/core_integrations/test.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315428/issues/source?bid=3282987&fileBranchId=3283570#l57)

File: core/core_integrations/test.js (hunk 29 above)
Comment (mrkno) [inline core/core_integrations/test.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315428/issues/source?bid=3282987&fileBranchId=3283570#l29)

File: core/integrations/hubot/robot.js [hunk 32]
```diff
@@ -0,0 +1,57 @@
...
+    userForId: function() {
+        return {
+            id: 'foo',
+            name: 'bar'
+        };
+    }
+};
+
+exports.shutdown = function () {
+    exports.logger.warning("Integrations should not invoke a safe shutdown. Please use the shutdown command itself.");
```
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315442/issues/source?bid=3282987&fileBranchId=3283570#l53)

File: core/integrations/hubot/robot.js [hunk 33]
```diff
@@ -0,0 +1,57 @@
+﻿var instance,
```
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315442/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/integrations/hubot/robot.js (hunk 32 above)
Comment (mrkno) [inline core/integrations/hubot/robot.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315442/issues/source?bid=3282987&fileBranchId=3283570#l53)

File: core/integrations/integrations.js (hunk 27 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315443/issues/source?bid=3282987&fileBranchId=3283570#l65)

File: core/integrations/integrations.js (hunk 28 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315443/issues/source?bid=3282987&fileBranchId=3283570#l1)

File: core/integrations/integrations.js (hunk 27 above)
Comment (mrkno) [inline core/integrations/integrations.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531315443/issues/source?bid=3282987&fileBranchId=3283570#l65)

**Pull request merged**